app/controllers/authenticate_controller.rb

```diff
@@ -3,6 +3,60 @@
 class AuthenticateController < ApplicationController
   include BasicAuthenticator
   include AuthorizeResource
+  include CurrentUser
+  include FindResource
+  include AssumedRole
+
+  def list_authenticators
+    # Rails 5 requires parameters to be explicitly permitted before converting
+    # to Hash.  See: https://stackoverflow.com/a/46029524
+    allowed_params = %i[account service_id]
+
+    begin
+      scope =  authenticators(
+        assumed_role(query_role),
+        repo = DB::Repository::AuthenticatorRepository.new,
+        handler = Authentication::Handler::OidcAuthenticationHandler.new,
+        **options(allowed_params)
+      )
+    rescue ApplicationController::Forbidden
+      raise
+    rescue ArgumentError => e
+      raise ApplicationController::UnprocessableEntity, e.message
+    end
+
+    render(json: scope)
+  end
+
+  def authenticators(role, repo, handler, account:, service_id: nil)
```

Comment:
Mind updating the positional arguments to be named arguments? They have the benefit of conveying context to the  calling code.
```suggestion
  def authenticators(role:, repository:, handler:, account:, service_id: nil)
```